src/app/timed-call/timed-call.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'timed-call',
+  templateUrl: './timed-call.component.html',
+  styleUrls: ['./timed-call.component.scss']
+})
+export class TimedCallComponent implements OnInit, OnDestroy {
+
+  // Function that will be called everytime the Timer reaches 0.
+  @Input()
+  functionToCall: Function;
```

Comment:
I think it would be more idiomatic if this was an `@Output` prop. That way, the timer just fires an event when it elapses and subscribers can respond how they see fit.

Consequently, I would drop the "call" from the component name and rename the component to something like e.g. IntervalTimer.